src/cargo/core/compiler/compilation.rs

```diff
@@ -82,18 +82,26 @@ pub struct Compilation<'cfg> {
 
 impl<'cfg> Compilation<'cfg> {
     pub fn new<'a>(bcx: &BuildContext<'a, 'cfg>) -> CargoResult<Compilation<'cfg>> {
-        let mut rustc = bcx.rustc.process();
+        // If we're using cargo as a rustc wrapper then we're in a situation
+        // like `cargo fix`. For now just disregard the `RUSTC_WRAPPER` env var
+        // (which is typically set to `sccache` for now). Eventually we'll
+        // probably want to implement `RUSTC_WRAPPER` for `cargo fix`, but we'll
+        // leave that open as a bug for now.
+        let mut rustc = if bcx.build_config.cargo_as_rustc_wrapper {
```

Comment:
This option sounds more generic than it actually is. Before looking it up I would've assumed it was also used for at least `cargo rustc`, maybe `cargo rustdoc` too (given the chat in #5958).

But, no, it's just set `ops::fix`.